quizapp/src/main/java/billennium/quizapp/service/QuizService.java

```diff
@@ -2,38 +2,95 @@
-            return mapToQuizDefinitionModel(queryResult.getQuizExecutedId(), queryResult.getQuiz());
+            return mapToQuizDefinitionModel(queryResult.getQuizExecutedId(), queryResult.getQuiz(), answersDto);
         } else {
             return new QuizDefinitionDto();
         }
     }
 
+    private QuizDefinitionDto mapToQuizDefinitionModel(Long quizExecutedId, QuizDefinition quiz,
+                                                       AnswersDto answersDTO) {
+        if (answersDTO.getQuestionId() == null) {
+            createQuizExecuted(quizExecutedId);
+            Question firstQuestion = getFirstQuestion(quiz);
+            return createQuizDefinitionDto(quizExecutedId, firstQuestion, quiz, firstQuestion.getId());
+        } else {
+            return goToNextQuestion(quizExecutedId, answersDTO, quiz);
+        }
+    }
+
+
+    private QuizDefinitionDto goToNextQuestion(Long quizExecutedId, AnswersDto answersDTO, QuizDefinition quiz) {
+        QuizExecuted quizExecuted = quizExecutedRepository.findById(quizExecutedId).orElseThrow(QuizExecutedException::new);
+        List<ResultDetails> resultDetails = saveGivenAnswer(answersDTO, quizExecuted);
+        return createQuizDefinitionDtoForNextQuestion(resultDetails, quiz, quizExecuted);
+    }
+
+    private List<ResultDetails> saveGivenAnswer(AnswersDto answersDTO, QuizExecuted quizExecuted) {
+        List<ResultDetails> resultDetails = quizExecuted.getResultDetails();
+        resultDetails.add(ResultDetails.builder()
+                .questionId(answersDTO.getQuestionId())
+                .givenAnswerId(answersDTO.getAnswerId())
+                .build());
+        quizExecuted.setResultDetails(resultDetails);
+        return resultDetails;
+    }
 
-    private QuizDefinitionDto mapToQuizDefinitionModel(Long quizExecutedId, QuizDefinition quiz) {
-        Question question = quiz.getQuestions().get(QUIZ_FIRST_QUESTION_NUMBER);
+    private QuizDefinitionDto createQuizDefinitionDtoForNextQuestion(List<ResultDetails> resultDetails,
+                                                                     QuizDefinition quiz, QuizExecuted quizExecuted) {
+        if (resultDetails.size() == quiz.getQuestions().size()) {
+            quizExecuted.setQuizStatus(QuizStatus.DONE);
+            return QuizDefinitionDto.builder().build();
+        } else {
+            Question nextQuestion = quiz.getQuestions().get(resultDetails.size());
+            return createQuizDefinitionDto(quizExecuted.getId(), nextQuestion, quiz, nextQuestion.getId());
+        }
+    }
+
+    private void createQuizExecuted(Long quizExecutedId) {
+        quizExecutedRepository.findById(quizExecutedId).ifPresent(quizExecuted ->
+                quizExecuted.setQuizStatus(QuizStatus.IN_PROGRESS)
+        );
+    }
+
+    private Question getFirstQuestion(QuizDefinition quiz) {
+        return quiz.getQuestions().stream().findFirst().orElseThrow(QuizDefinitionException::new);
```

Comment:
niepotrzebnie w nowej metodzie. kod sam z siebie tlumaczy ze bierzemy pierwszy obiekt albo zwracamy excepction

Reply:
ok ale dla czytelności warto by tak było w parametrze metody np. było by to dużo mniej czytelne.

Reply:
chociaz dobra przypisze to po prostu do zmiennej i tyle :)